File: api-tester/src/main/java/com/revenuecat/apitester/java/DeepLinkHandlerAPI.java
package com.revenuecat.apitester.java;

import android.content.Intent;

import androidx.annotation.OptIn;

import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI;
import com.revenuecat.purchases.deeplinks.DeepLinkHandler;

@OptIn(markerClass = ExperimentalPreviewRevenueCatPurchasesAPI.class)
@SuppressWarnings({"unused"})
final class DeepLinkHandlerAPI {
    static void check(Intent intent, Boolean shouldCache) {
        DeepLinkHandler.Result result = DeepLinkHandler.handleDeepLink(intent, shouldCache);
    }

    static boolean checkResult(DeepLinkHandler.Result result) {
        switch (result) {
            case HANDLED:
            case IGNORED:
            case DEFERRED:
                return true;
            default:
                return false;
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/DeepLinkHandlerAPI.kt
package com.revenuecat.apitester.kotlin

import android.content.Intent
import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
import com.revenuecat.purchases.deeplinks.DeepLinkHandler

@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
@Suppress("unused", "UNUSED_VARIABLE")
private class DeepLinkHandlerAPI {
    fun check(intent: Intent, shouldCache: Boolean) {
        val result: DeepLinkHandler.Result = DeepLinkHandler.handleDeepLink(intent, shouldCache)
    }

    fun checkHandleResult(result: DeepLinkHandler.Result): Boolean {
        when (result) {
            DeepLinkHandler.Result.HANDLED,
            DeepLinkHandler.Result.DEFERRED,
            DeepLinkHandler.Result.IGNORED,
            -> {
                return true
            }
        }
    }
}
